homeassistant/components/binary_sensor/zha.py

```diff
@@ -102,3 +137,114 @@ def cluster_command(self, tsn, command_id, args):
         state = result.get('zone_status', self._state)
         if isinstance(state, (int, uint16_t)):
             self._state = result.get('zone_status', self._state) & 3
+
+
+class Switch(zha.Entity, BinarySensorDevice):
+    """ZHA switch/remote controller/button."""
+
+    _domain = DOMAIN
+
+    class OnOffListener:
```

Comment:
Why nest the listener classes?

Reply:
The idea was that they are pretty tightly bound to the Switch class, and are only classes because of the zigpy API. *shrug*